### PR TITLE
Removed PodSecurityPolicy from deployments to support K8s 1.25+

### PR DESCRIPTION
**Pull request by gauravAggrahari**
Signed-off-by: Gaurav Aggrahari <gaurav.kumar.aggrahari@ibm.com>

### CODE REVIEW

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json [hunk 1]
```diff
@@ -33,10 +33,6 @@
             "name": "supported-envs",
             "value": "Google"
         },
-        {
-            "name": "supported-k8s-versions",
-            "value": "1.22, 1.23, 1.24, 1.25"
-        },
         {
             "name": "supported-ocp-versions",
             "value": "4.8, 4.9, 4.10, 4.11"
```
Comment (anandkarra) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: The `supported-ocp-versions` section is the one to be removed here.

**Review by anandkarra — COMMENTED** [review]
Added comments

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json (hunk 1 above)
Comment (gauravAggrahari) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json]: aah yes. Good catch. Thank you.

**Review by anandkarra — APPROVED** [review]
LGTM

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.0.4/metadata.json [hunk 2]
```diff
@@ -37,10 +37,6 @@
             "name": "supported-k8s-versions",
             "value": "1.18, 1.19, 1.20"
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.0.4/metadata.json]: Lets update k8s versions from 1.21 to 1.25 though its not being used in code logic but just for user's information purpose

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.7.1/deployment.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: kube-system
   annotations:
     version: 1.7.1
-    revision: 4
+    revision: 5
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.7.1/deployment.yaml:9]: I thik we can remove this line also
https://github.com/IBM/ibm-satellite-storage/blob/4646fb74a5f2cb06c662d6f46f283b8d4716fe72/config-templates/google/gcp-compute-persistent-disk-csi-driver/1.7.1/deployment.yaml#L791-L793

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.7.1/metadata.json [hunk 4]
```diff
@@ -37,10 +37,6 @@
             "name": "supported-k8s-versions",
             "value": "1.20, 1.21, 1.22, 1.23"
         },
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.7.1/metadata.json:39]: Same here

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml [hunk 5]
```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: kube-system
   annotations:
     version: 1.8.0
-    revision: 1
+    revision: 2
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml:9]: Same here https://github.com/IBM/ibm-satellite-storage/blob/4646fb74a5f2cb06c662d6f46f283b8d4716fe72/config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/deployment.yaml#L793-L795

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json [hunk 6]
```diff
@@ -37,10 +37,6 @@
             "name": "supported-k8s-versions",
             "value": "1.22, 1.23, 1.24, 1.25"
         },
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/1.8.0/metadata.json:39]: include 1.21 as well

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/changelog.json [hunk 7]
```diff
@@ -35,6 +57,17 @@
         "1.0.4": {
             "status": "supported",
             "latest-revision": {
+                "revision": "4",
+                "release-date": "16 January 2023",
```
Comment (mssachan) [inline config-templates/google/gcp-compute-persistent-disk-csi-driver/changelog.json]: Change release date to 23rd Jan

**Review by mssachan — CHANGES_REQUESTED** [review]
Lets remove all resources related with PSP including ClusterRoles and ClusterRoleBindings whichever is applicable

File: config-templates/google/gcp-compute-persistent-disk-csi-driver/1.0.4/metadata.json (hunk 2 above)
Comment (gauravAggrahari) [reply config-templates/google/gcp-compute-persistent-disk-csi-driver/1.0.4/metadata.json]: These versions is based on CSI driver support. I guess they supported only these while developing this template

**Comment by mssachan** [comment]
Update msg `Removed PodSecurityPolicy as Kubernetes v1.25+ does not support PodSecurityPolicy`

**Review by mssachan — APPROVED** [review]
lgtm

**Pull request merged**